### PR TITLE
Replace log with tracing

### LINKED ISSUE #3482: Use tracing instead of log
`tracing` is generally used more than `log` in modern Rust

### LINKED ISSUE #3488: Outdated err message cites `EventLoopBuilderExtUnix::any_thread`
Using `egui`+`winit` I got an error message citing `EventLoopBuilderExtUnix::any_thread`. I believe this error message is outdated -- it seems `EventLoopBuilderExtUnix` has been replaced with e.g. `EventLoopBuilderExtX11`.

<details>
<summary>
The error message
</summary>

```
thread '<unnamed>' panicked at /home/homestar/dev/mrepo/src_rs/trading/.vendor/winit/src/platform_impl/linux/mod.rs:747:13:
Initializing the event loop outside of the main thread is a significant cross-platform compatibility hazard. If you absolutely need to create an EventLoop on a different thread, you can use the `EventLoopBuilderExtUnix::any_thread` function.
stack backtrace:
   0: rust_begin_unwind
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/panicking.rs:72:14
   2: winit::platform_impl::platform::EventLoop<T>::new
   3: eframe::native::run::create_event_loop
   4: eframe::native::run::run_glow
   5: eframe::run_native
   6: trading_gui::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
</details>

The error message originates from here: https://github.com/rust-windowing/winit/blob/6825fed07379b942e6a8df94f7ee16514d5f7e7c/src/platform_impl/linux/mod.rs#L740-L745

### PR DESCRIPTION
**Pull request by notgull**
Tracing is a modern replacement for the log crate that allows 
for annotating log messages with the function that they come from.

Closes #3482

- [ ] Tested on all platforms changed
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Review by madsmtm — COMMENTED** [review]
How does this work with `simple_logger` that we use in examples?

**Comment by notgull** [comment]
> How does this work with `simple_logger` that we use in examples?

If you enable the `log` feature it will use the logger. However for convenience I've replaced it with `tracing-subscriber`.

File: examples/x11_embed.rs [hunk 1]
```diff
@@ -21,7 +19,7 @@ fn main() -> Result<(), Box<dyn Error>> {
         .ok_or("Expected a 32-bit X11 window ID as the first argument.")?
         .parse::<u32>()?;
 
-    SimpleLogger::new().init().unwrap();
+    tracing_subscriber::fmt::init();
     let event_loop = EventLoop::new()?;
 
     let mut window = None;
```
Comment (madsmtm) [inline examples/x11_embed.rs:25]: Nit: Re-add logging to `window` example

File: Cargo.toml [hunk 2]
```diff
@@ -64,17 +64,17 @@ cfg_aliases = "0.2.0"
...
 rwh_06 = { package = "raw-window-handle", version = "0.6", features = ["std"], optional = true }
 serde = { workspace = true, optional = true }
 smol_str = "0.2.0"
 dpi = { path = "dpi" }
+tracing = { version = "0.1.40", default_features = false }
 
 [dev-dependencies]
 image = { version = "0.24.0", default-features = false, features = ["png"] }
-simple_logger = { version = "4.2.0", default_features = false }
+tracing-subscriber = "0.3.18"
```
Comment (madsmtm) [inline Cargo.toml:77]: Nit: Enable filtering with `RUST_LOG`?

```suggestion
tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
```

**Review by madsmtm — APPROVED** [review]

File: src/error.rs [hunk 3]
```diff
@@ -65,7 +65,9 @@ impl OsError {
 #[allow(unused_macros)]
 macro_rules! os_error {
     ($error:expr) => {{
-        crate::error::OsError::new(line!(), file!(), $error)
+        let error = $error;
+        tracing::error!("encountered an OS error: {}", error);
```
[the remark this answers is not in the document]
Comment (notgull) [reply src/error.rs]: Okay, I've removed it for now

**Review by kchibisov — APPROVED** [review]

**Pull request merged**